dashboard/src/routes/dashboard/index.svelte

```diff
@@ -0,0 +1,152 @@
+<script lang="ts">
+	import { createDevice, getDevices } from '$lib';
+	import type { Device } from '$lib/types';
+	import {
+		authenticationData,
+		Icon,
+		ListManager,
+		NotificationManager,
+		type ActionButton,
+		type TableData
+	} from 'boxfish-studio--is-ui-components';
+	import { onMount } from 'svelte';
+	import { Container, Row } from 'sveltestrap';
+	import { DeviceDetails, ProgressBar } from '../../components';
+	import { progress } from '$lib/store';
+
+	enum State {
+		ListDevices = 'listDevices',
+		DeviceDetails = 'deviceDetails'
+	}
+
+	let devices = [];
+	let loading: boolean = false;
+	let state: State = State.ListDevices;
+	let message: string;
+	let selectedDevice: Device;
+	let query: string;
+	let searchResults: Device[] = [];
+	let CREATE_DEVICE_BUTTON: ActionButton = {
```

Comment:
if this is a variable it shouldnt be written in `Screaming Snake Case`. I would write it:

```ts
let createDeviceButton: ActionButton
$: createDeviceButton = {
    onClick: handleCreateDevice,
    icon: 'plus',
    color: 'dark',
    label: loading ? 'Creating device...' : 'Create device',
    loading,
    disabled: loading
};
```